src/viam/sdk/services/mlmodel/mlmodel.hpp

```diff
@@ -111,6 +111,17 @@ class MLModelService : public Service {
     /// viewed by the views is only valid for the lifetime of the returned shared_ptr.
     virtual std::shared_ptr<named_tensor_views> infer(const named_tensor_views& inputs) = 0;
 
+    /// @brief Runs the model against the input tensors and returns inference results as tensors.
+    ///
+    /// @param `inputs`: The input tensors on which to run inference.
+    ///
+    /// @param `extra`: Any additional arguments to the method.
+    ///
+    /// @returns The results of the inference as a shared pointer to `named_tensor_views`. The data
+    /// viewed by the views is only valid for the lifetime of the returned shared_ptr.
+    virtual std::shared_ptr<named_tensor_views> infer(const named_tensor_views& inputs,
```

Comment:
- I think the tflite example and audio client will need an update.

- I don't love that this requires all callers to now provide an extra argument when 99% of the time it will go unused. It is generally not a good idea to mix defaulted parameters with virtual functions because you are combining static and dynamic binding. But you could consider adding a non-virtual overload:
```
inline std::shared_ptr<named_tensor_views> infer(const named_tensor_views& inputs) {
    return infer(inputs, {});
}

virtual std::shared_ptr<named_tensor_views> infer(const named_tensor_views& inputs, const AttributeMap& extra);
```

Or just decide that you aren't that worried about adding the defaults on a virtual and just give `extra` a defaulted `={}`.

Same thoughts, really, for all the API methods that have the new `extra` parameter.

Reply:
(discussed offline, will de-virtualize the `extra`less method overloads)